pages/settlement/[id].tsx

```diff
@@ -1,140 +1,203 @@
+        throw new Error('Contract is not authorised')
+      }
+      const tx = STL.setApprovalForAll(SETTLEMENT_V2_CONTRACT_ADDRESS, true);
+      await handleTx(tx);
+      await revalidateApproval()
+    } catch (e) {
+      console.error(e)
+    }
+  }, [STL, STLV2, handleTx, isReadOnly, revalidateApproval])
+
+  const migrateV2 = useCallback(async () => {
+    try {
+      if (!STL || !STLV2 || isReadOnly) {
+        throw new Error('Contract is not authorised')
+      }
+      const dto = await buildMigrationPayload(id, STL);
+      console.log(STLV2)
+      const tx = STLV2.claim(
+        id,
+        [dto.size,
+          dto.spirit,
+          dto.age,
+          dto.resource,
+          dto.morale,
+          dto.government,
+          dto.turns]
+      )
+      await handleTx(tx)
+    } catch (err) {
+      console.log(err)
     }
-    await handleTx(STL.randomise(id))
-  }, [STL, handleTx, id, isReadOnly])
+  }, [STL, STLV2, handleTx, id, isReadOnly])
 
+  const isOwner = useMemo(() => owner && account && isAddressMatch(account, owner), [account, owner])
 
   const rolls = useMemo(() => {
-    const realm = initialData.nft.tokenData.metadata.json.attributes[6].value;
+    const realm = data?.attributes[6].value;
     return 5 - realms.indexOf(realm) || 0
-  }, [initialData.nft.tokenData.metadata.json.attributes])
+  }, [data])
 
-
-  if (isFallback || !initialData) {
+  if (isFallback || !data) {
     return <p>
       loading...
     </p>
   }
 
+
   return (
     <Fragment>
       <Head>
-        <title>{initialData?.name}</title>
+        <title>{data?.name}</title>
         <meta name="description" content={description}/>
       </Head>
       <Header/>
       <main className={styles.main}>
-        <h1>{initialData.name}</h1>
+        <h1>{data.name}</h1>
         <NFTFullPage
           useBetaIndexer={true}
           contract={SETTLEMENT_CONTRACT_ADDRESS}
           id={id}
         >
           <div className={styles.media}>
-            <MediaObject metadata={{...initialData.nft.tokenData.metadata.json, image}}/>
+            <MediaObject metadata={{...data, mimeType: 'image/svg+xml'}}/>
           </div>
-          {account && isAddressMatch(account, initialData.nft.tokenData?.owner) && (
+
+          {isOwner && (
+            <div className={styles.manageContainer}>
+              {legacy ?
+                <ManageLegacySettlement
+                  txInProgress={txInProgress}
+                  txStatus={txStatus}
+                  approved={!!approved}
+                  approve={approveV2}
+                  migrate={migrateV2}
+                /> :
+                <ManageV2Settlement id={id} rolls={rolls}/>
+              }
+            </div>
+          )}
+          {account && (
+            <div>
+              <h3>Your Resource Balances</h3>
+              {resources.map(r => <ResourceBalance key={r} resource={r}/>)}
+            </div>
+          )}
+          {NETWORK_CHAIN_ID === 1 && (
```

Comment:
Idk what this is but is it 100% going to work with V2 on mainnet? Just asking coz i dont think we would have tested this on rinkeby because it wouldnt have rendered in.

Reply:
ah its zora indexer components to fetch minter vs owner, it'll work but ill nix for now